annotation-processor/src/main/java/org/hibernate/validator/ap/internal/checks/TypeCheck.java

```diff
@@ -60,25 +69,60 @@ public TypeCheck(ConstraintHelper constraintHelper) {
 	}
 
 	@Override
-	public Set<ConstraintCheckIssue> checkNonAnnotationType(
-			TypeElement element, AnnotationMirror annotation) {
-
+	public Set<ConstraintCheckIssue> checkNonAnnotationType(TypeElement element, AnnotationMirror annotation) {
 		return checkInternal( element, annotation, element.asType(), "NOT_SUPPORTED_TYPE" );
 	}
 
-	private Set<ConstraintCheckIssue> checkInternal(Element element,
-			AnnotationMirror annotation, TypeMirror type, String messageKey) {
-
-		if ( constraintHelper.checkConstraint(
-				annotation.getAnnotationType(), type ) != ConstraintCheckResult.ALLOWED ) {
-
+	private Set<ConstraintCheckIssue> checkInternal(Element element, AnnotationMirror annotation, TypeMirror type, String messageKey) {
+		TypeMirror typeToCheck = usesUnwrapping( annotation, type ) ? getUnwrappedType( type ) : type;
+		if ( !isAnnotationAllowedForType( annotation, typeToCheck ) ) {
 			return CollectionHelper.asSet(
 					ConstraintCheckIssue.error(
 							element, annotation, messageKey,
-							annotation.getAnnotationType().asElement().getSimpleName() ) );
+							annotation.getAnnotationType().asElement().getSimpleName()
+					) );
 		}
 
 		return Collections.emptySet();
 	}
 
+	private boolean isAnnotationAllowedForType(AnnotationMirror annotation, TypeMirror type) {
+		return ConstraintCheckResult.ALLOWED.equals( constraintHelper.checkConstraint( annotation.getAnnotationType(), type ) );
+	}
+
+	private boolean usesUnwrapping(AnnotationMirror annotationMirror, TypeMirror typeMirror) {
+		//need to check if this annotation is not on one of the types that are automatically unwrapped:
+		if ( constraintHelper.isSupportedForUnwrappingByDefault( getQualifiedName( typeMirror ) ) ) {
+			return true;
```

Comment:
We have to take into account the case where it's unwrapped by default BUT we have the `Skip` payload. In this case, it doesn't use unwrapping.